### PR TITLE
No Cheerio in the client bundle

### PR DESCRIPTION
**Pull request by jimrandomh**
Saves some JS bundle size by not using Cheerio (which is only supposed to be a server-side lib). Adds a resolver-only field to handle the one thing the client was legitimately using Cheerio for. Adds a lint rule (and template for client-restricting things in lint in general.)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206568113662608) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/resolvers/postResolvers.ts [hunk 1]
```diff
@@ -288,6 +288,32 @@ augmentFieldsDict(Posts, {
...
+  firstVideoAttribsForPreview: {
+    resolveAs: {
+      type: GraphQLJSON,
+      resolver: async (post: DbPost, args: void, context: ResolverContext) => {
+        const videoHosts = [
+          "https://www.youtube.com",
+          "https://youtube.com",
+          "https://youtu.be",
+        ];
+        const $ = cheerioParse(post.contents.html);
```
Comment (s-cheng) [inline packages/lesswrong/server/resolvers/postResolvers.ts:301]: BTW `post.contents` can be missing, like for a crosspost hosted on another site

**Review by s-cheng — APPROVED** [review]
Nice thanks! ASTLAC

**Comment by Discordius** [comment]
@jimrandomh You probably want to just merge this

**Comment by s-cheng** [comment]
Please don't merge this without fixing the issue I pointed out, because I think it will break something on our "Best of" page

**Comment by Discordius** [comment]
@jimrandomh Would be good to wrap this up and merge it

**Pull request merged**